Build fails with Docker image on MacOS

_For new Package Requests, see the guidelines_  

### Setup
_Package Name:_ git
_Package Version:_

_NAS Model:_ ds418
_NAS Architecture:_ rtd1296
_DSM version:_

### Expected behavior
make arch-rtd1296
Make succeeds

### Actual behavior


### Steps to reproduce
_1._ Install docker.
_2._ Run docker environment
_3._ Git clone the tree
_4._ cd spk/git
_5._ make arch-rtd1296


#### Package log
_Check Package Center or `/usr/local/{package}/var/`_
```
root@f6dcda86815a:/spksrc/spksrc/spk/git# make arch-rtd1296
===>  Building package for arch rtd1296
make[1]: Entering directory '/spksrc/spksrc/spk/git'
===>  Set up toolchain 
===>    Extracting for syno-rtd1296
tar -xpf /spksrc/spksrc/toolchains/syno-rtd1296-6.1/../../distrib/toolchains/6.1/rtd1296-gcc494_glibc220_armv8-GPL.txz -C /spksrc/spksrc/toolchains/syno-rtd1296-6.1/work 
tar: aarch64-unknown-linux-gnueabi/build.log.bz2: Cannot open: Permission denied
tar: aarch64-unknown-linux-gnueabi/aarch64-unknown-linux-gnueabi: Cannot mkdir: Permission denied
tar: aarch64-unknown-linux-gnueabi/aarch64-unknown-linux-gnueabi: Cannot mkdir: Permission denied
...
```
#### Other logs
_E.g. `/var/log/messages` or `/var/log/synopkg.log`_
```
Insert log here
```

This is the result of running the same test on a RHEL linux - docker CE.

$ docker pull synocommunity/spksrc                                                                                                     
Using default tag: latest                                                                                                                                   
latest: Pulling from synocommunity/spksrc                                                                                                                   
ff144d3c0ab1: Pull complete                                                                                                                                 
a6429e90a8e7: Pull complete                                                                                                                                 
5a1b738ba18b: Pull complete                                                                                                                                 
be42538d7299: Pull complete                                                                                                                                 
0c929307a750: Pull complete                                                                                                                                 
a7db208018b5: Pull complete                                                                                                                                 
7095ea3d03f9: Pull complete                                                                                                                                 
Digest: sha256:3b1be74859df976c927e6b5834ed2ec9f35eea36599c2272e50849cd703b9515                                                                             
Status: Downloaded newer image for synocommunity/spksrc:latest                                                                                              
$ mkdir spksrc                                                                                                                         
$ docker run -it -v ~/spksrc:/spksrc:rw synocommunity/spksrc /bin/bash                                                                 
root@ffdba332b86d:/spksrc# git clone https://github.com/SynoCommunity/spksrc.git                                                                            
Cloning into 'spksrc'...                                                                                                                                    
remote: Enumerating objects: 61, done.                                                                                                                      
remote: Counting objects: 100% (61/61), done.                                                                                                               
remote: Compressing objects: 100% (40/40), done.                                                                                                            
remote: Total 29314 (delta 27), reused 45 (delta 20), pack-reused 29253                                                                                     
Receiving objects: 100% (29314/29314), 12.22 MiB | 2.88 MiB/s, done.                                                                                        
Resolving deltas: 100% (15186/15186), done.                                                                                                                 
Checking connectivity... done.                                                                                                                              
root@ffdba332b86d:/spksrc# cd spksrc/                                                                                                                       
root@ffdba332b86d:/spksrc/spksrc# make setup                                                                                                                
k/transmissionCreating local configuration "local.mk"...                                                                                                    
Setting default toolchain version to DSM-6.1                                                                                                                
root@ffdba332b86d:/spksrc/spksrc# cd spk/transmission                                                                                                       
root@ffdba332b86d:/spksrc/spksrc/spk/transmission# make arch-rtd296                                                                                         
===>  Building package for arch rtd296                                                                                                                      
make[1]: Entering directory '/spksrc/spksrc/spk/transmission'                                                                                               
===>  Set up toolchain                                                                                                                                      
make[2]: Entering directory '/spksrc/spksrc/spk/transmission'                                                                                               
make[2]: *** ../../toolchains/syno-rtd296-6.1: No such file or directory.  Stop.                                                                            
make[2]: Leaving directory '/spksrc/spksrc/spk/transmission'                                                                                                
/spksrc/spksrc/spk/transmission/work-rtd296-6.1/tc_vars.mk:1: *** An error occured while setting up the toolchain, please check the messages above.  Stop.  
make[1]: Leaving directory '/spksrc/spksrc/spk/transmission'                                                                                                
../../mk/spksrc.spk.mk:452: recipe for target 'arch-rtd296' failed                                                                                          
make: [arch-rtd296] Error 2 (ignored)                                                                                                                       
root@ffdba332b86d:/spksrc/spksrc/spk/transmission#  make arch-88f6281
===>  Building package for arch 88f6281
make[1]: Entering directory '/spksrc/spksrc/spk/transmission'
===>  Set up toolchain
===>    Downloading files for syno-88f6281
wget https://sourceforge.net/projects/dsgpl/files/DSM%206.1%20Tool%20Chains/Marvell%2088F628x%20Linux%202.6.32/6281-gcc464_glibc215_88f6281-GPL.txz
2019-01-10 00:10:04 URL:https://pilotfiber.dl.sourceforge.net/project/dsgpl/DSM%206.1%20Tool%20Chains/Marvell%2088F628x%20Linux%202.6.32/6281-gcc464_glibc215_88f6281-GPL.txz [35293008/35293008] -> "6281-gcc464_glibc215_88f6281-GPL.txz.part" [1] make arch-88f6281
===>  Building package for arch 88f6281
make[1]: Entering directory '/spksrc/spksrc/spk/transmission'
===>  Set up toolchain
===>    Downloading files for syno-88f6281
wget https://sourceforge.net/projects/dsgpl/files/DSM%206.1%20Tool%20Chains/Marvell%2088F628x%20Linux%202.6.32/6281-gcc464_glibc215_88f6281-GPL.txz
2019-01-10 00:10:04 URL:https://pilotfiber.dl.sourceforge.net/project/dsgpl/DSM%206.1%20Tool%20Chains/Marvell%2088F628x%20Linux%202.6.32/6281-gcc464_glibc215_88f6281-GPL.txz [35293008/35293008] -> "6281-gcc464_glibc215_88f6281-GPL.txz.part" [1]
...

--- a/Dockerfile
+++ b/Dockerfile
@@ -20,6 +20,7 @@ RUN apt-get update && apt-get install --no-install-recommends -y \
 		debootstrap \
 		ed \
 		expect \
+		fakeroot \
 		flex \
 		g++-multilib \
 		gawk \

--- a/README.rst
+++ b/README.rst
@@ -32,7 +32,9 @@ Docker
 * `Fork and clone`_ spksrc: ``git clone https://github.com/YOUR-USERNAME/spksrc ~/spksrc``
 * Install Docker on your host OS: `Docker installation`_. A wget-based alternative for linux: `Install Docker with wget`_.
 * Download the spksrc docker container: ``docker pull synocommunity/spksrc``
-* Run the container with ``docker run -it -v ~/spksrc:/spksrc synocommunity/spksrc /bin/bash``
+* Run the container with ``docker run -it -v ~/spksrc:/spksrc synocommunity/spksrc /bin/bash``. 
+  On Mac hosts running OSX you might need to use fakeroot with tar to avoid permission errors while extracting the synology toolchain archives.
+  For that reason run the container with ``docker run -it -v~/spksrc:/spksrc -e TAR_CMD="fakeroot tar" synocommunity/spksrc /bin/bash``.
 
 
 Virtual machine

--- a/mk/spksrc.extract.mk
+++ b/mk/spksrc.extract.mk
@@ -8,7 +8,7 @@
 #  post_extract_target  (override with POST_EXTRACT_TARGET)
 EXTRACT_PATH ?= $(WORK_DIR)
 
-# Extract commands
+# Extract commands (optionally use TAR_CMD="fakeroot tar" as workaround on OSX)
 TAR_CMD ?= tar
 ifeq ($(strip $(EXTRACT_CMD.$(DIST_EXT))),)
 EXTRACT_CMD.tgz = $(TAR_CMD) -xzpf $(DIST_FILE) -C $(EXTRACT_PATH)

--- a/mk/spksrc.extract.mk
+++ b/mk/spksrc.extract.mk
@@ -7,16 +7,18 @@
 #  extract_target       (override with EXTRACT_TARGET)
 #  post_extract_target  (override with POST_EXTRACT_TARGET)
 EXTRACT_PATH ?= $(WORK_DIR)
+
 # Extract commands
+TAR_CMD = fakeroot tar
 ifeq ($(strip $(EXTRACT_CMD.$(DIST_EXT))),)
-EXTRACT_CMD.tgz = tar -xzpf $(DIST_FILE) -C $(EXTRACT_PATH)
-EXTRACT_CMD.txz = tar -xpf $(DIST_FILE) -C $(EXTRACT_PATH)
-EXTRACT_CMD.tar.gz = tar -xzpf $(DIST_FILE) -C $(EXTRACT_PATH)
-EXTRACT_CMD.tar.bz2 = tar -xjpf $(DIST_FILE) -C $(EXTRACT_PATH)
-EXTRACT_CMD.tbz = tar -xjpf $(DIST_FILE) -C $(EXTRACT_PATH)
-EXTRACT_CMD.tar.xz = tar -xJpf $(DIST_FILE) -C $(EXTRACT_PATH)
-EXTRACT_CMD.tar.lzma = tar --lzma -xpf $(DIST_FILE) -C $(EXTRACT_PATH)
-EXTRACT_CMD.tar.lz = tar --lzip -xpf $(DIST_FILE) -C $(EXTRACT_PATH)
+EXTRACT_CMD.tgz = $(TAR_CMD) -xzpf $(DIST_FILE) -C $(EXTRACT_PATH)
+EXTRACT_CMD.txz = $(TAR_CMD) -xpf $(DIST_FILE) -C $(EXTRACT_PATH)
+EXTRACT_CMD.tar.gz = $(TAR_CMD) -xzpf $(DIST_FILE) -C $(EXTRACT_PATH)
+EXTRACT_CMD.tar.bz2 = $(TAR_CMD) -xjpf $(DIST_FILE) -C $(EXTRACT_PATH)
+EXTRACT_CMD.tbz = $(TAR_CMD) -xjpf $(DIST_FILE) -C $(EXTRACT_PATH)
+EXTRACT_CMD.tar.xz = $(TAR_CMD) -xJpf $(DIST_FILE) -C $(EXTRACT_PATH)
+EXTRACT_CMD.tar.lzma = $(TAR_CMD) --lzma -xpf $(DIST_FILE) -C $(EXTRACT_PATH)
+EXTRACT_CMD.tar.lz = $(TAR_CMD) --lzip -xpf $(DIST_FILE) -C $(EXTRACT_PATH)
 EXTRACT_CMD.zip = unzip $(DIST_FILE) -d $(EXTRACT_PATH)
 EXTRACT_CMD.deb = dpkg-deb -R $(DIST_FILE) $(EXTRACT_PATH)
 endif

--- a/mk/spksrc.extract.mk
+++ b/mk/spksrc.extract.mk
@@ -9,7 +9,7 @@
 EXTRACT_PATH ?= $(WORK_DIR)
 
 # Extract commands
-TAR_CMD = fakeroot tar
+TAR_CMD ?= tar
 ifeq ($(strip $(EXTRACT_CMD.$(DIST_EXT))),)
 EXTRACT_CMD.tgz = $(TAR_CMD) -xzpf $(DIST_FILE) -C $(EXTRACT_PATH)
 EXTRACT_CMD.txz = $(TAR_CMD) -xpf $(DIST_FILE) -C $(EXTRACT_PATH)

